gobblin-data-management/src/main/java/gobblin/data/management/copy/CopySource.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.SourceState;
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.data.management.copy.extractor.InputStreamExtractor;
+import gobblin.data.management.dataset.Dataset;
+import gobblin.data.management.dataset.DatasetUtils;
+import gobblin.data.management.partition.Partition;
+import gobblin.data.management.retention.dataset.finder.DatasetFinder;
+import gobblin.source.extractor.Extractor;
+import gobblin.source.extractor.extract.AbstractSource;
+import gobblin.source.workunit.Extract;
+import gobblin.source.workunit.WorkUnit;
+import gobblin.util.HadoopUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.collect.Lists;
+
+
+/**
+ * {@link gobblin.source.Source} that generates work units from {@link gobblin.data.management.copy.CopyableDataset}s.
+ *
+ */
+public class CopySource extends AbstractSource<String, FileAwareInputStream> {
+
+  public static final String DEFAULT_DATASET_PROFILE_CLASS_KEY = CopyableGlobDatasetFinder.class.getCanonicalName();
```

Comment:
Dataset profile was intended to include dataset finder + default configurations for specific types of dataset (e.g. DatabaseSnapshot, or TrackingData). With the whole configuration management stuff, it might be less relevant. As features evolved (e.g. around IDPC and Kafka) they became almost synonymous.
Since the cleaner uses "profile.class", I think we should stick with profile for configuration keys.